examples/datasource-http-backend/pkg/plugin/datasource.go

```diff
@@ -98,24 +113,29 @@ func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, quer
 		return response, fmt.Errorf("new request with context: %w", err)
 	}
 	resp, err := d.httpClient.Do(req)
-	if err != nil {
-		return response, fmt.Errorf("http client do: %w", err)
+	switch {
+	case errors.Is(err, context.DeadlineExceeded):
+		return response, err
```

Comment:
In order to communicate it more clearly, I would either do 

```go
return nil, err
```

Or 

```go
return backend.DataResponse{}, err
```

as at first glance it can be confusing when we're handling two "response" variables in this function.